File: backend/hackathon/src/main/java/com/dev_trails/hackathon/FraudLog.java
package com.dev_trails.hackathon;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Column;

@Entity
public class FraudLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String claimId;
    public Long riderId;
    public String riderName;
    public Boolean fraudFlag;
    public Double confidenceScore;
    public Double fraudScore;
    public String mlPrediction;
    @Column(length = 1000)
    public String fraudReasons;  // comma-separated reasons
    public String zone;
    public Double gpsLat;
    public Double gpsLon;
    public Double networkRttMs;
    public String timestamp;
    public String verdict;  // BLOCKED | PASSED | FLAGGED
}

File: backend/models/FraudLog.js
const mongoose = require('mongoose');

const FraudLogSchema = new mongoose.Schema({
  claim_id: { type: String },
  rider_id: { type: mongoose.Schema.Types.Mixed },
  rider_name: { type: String },
  fraud_flag: { type: Boolean, default: false },
  confidence_score: { type: Number },
  fraud_score: { type: Number },
  ml_prediction: { type: String },
  fraud_reasons: { type: String },
  zone: { type: String },
  gps_lat: { type: Number },
  gps_lon: { type: Number },
  network_rtt_ms: { type: Number },
  timestamp: { type: String },
  verdict: { type: String },
}, { timestamps: false });

FraudLogSchema.set('toJSON', { virtuals: true });
FraudLogSchema.virtual('id').get(function () { return this._id.toHexString(); });

module.exports = mongoose.model('FraudLog', FraudLogSchema);
